bridges/snowbridge/pallets/inbound-queue-v2/src/lib.rs

```diff
@@ -0,0 +1,278 @@
+			}
+		}
+	}
+
+	/// The nonce of the message been processed or not
+	#[pallet::storage]
+	pub type NonceBitmap<T: Config> = StorageMap<_, Twox64Concat, u128, u128, ValueQuery>;
+
+	/// The current operating mode of the pallet.
+	#[pallet::storage]
+	pub type OperatingMode<T: Config> = StorageValue<_, BasicOperatingMode, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Submit an inbound message originating from the Gateway contract on Ethereum
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::submit())]
+		pub fn submit(origin: OriginFor<T>, message: Message) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+			ensure!(!OperatingMode::<T>::get().is_halted(), Error::<T>::Halted);
+
+			// submit message to verifier for verification
+			T::Verifier::verify(&message.event_log, &message.proof)
+				.map_err(|e| Error::<T>::Verification(e))?;
+
+			// Decode event log into an Envelope
+			let envelope =
+				Envelope::try_from(&message.event_log).map_err(|_| Error::<T>::InvalidEnvelope)?;
+
+			// Verify that the message was submitted from the known Gateway contract
+			ensure!(T::GatewayAddress::get() == envelope.gateway, Error::<T>::InvalidGateway);
+
+			// Verify the message has not been processed
+			ensure!(!Nonce::<T>::get(envelope.nonce.into()), Error::<T>::InvalidNonce);
+
+			let origin_account_location = Self::account_to_location(who)?;
+
+			let (xcm, _relayer_reward) =
+				Self::do_convert(envelope.message, origin_account_location.clone())?;
+
+			// Todo: Deposit fee(in Ether) to RewardLeger which should cover all of:
+			// T::RewardLeger::deposit(who, relayer_reward.into())?;
+			// a. The submit extrinsic cost on BH
+			// b. The delivery cost to AH
+			// c. The execution cost on AH
+			// d. The execution cost on destination chain(if any)
+			// e. The reward
+
+			// Set nonce flag to true
+			log::info!(target: "snowbridge-inbound-queue:v2","💫 setting nonce to {:?}", envelope.nonce);
+			Nonce::<T>::set(envelope.nonce.into());
+
+			// Attempt to forward XCM to AH
+			let message_id = Self::send_xcm(xcm, T::AssetHubParaId::get())?;
+			Self::deposit_event(Event::MessageReceived { nonce: envelope.nonce, message_id });
+
+			Ok(())
+		}
+
+		/// Halt or resume all pallet operations. May only be called by root.
+		#[pallet::call_index(1)]
+		#[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
+		pub fn set_operating_mode(
+			origin: OriginFor<T>,
+			mode: BasicOperatingMode,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			OperatingMode::<T>::set(mode);
+			Self::deposit_event(Event::OperatingModeChanged { mode });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn account_to_location(account: AccountIdOf<T>) -> Result<Location, Error<T>> {
+			let account_bytes: [u8; 32] =
+				account.encode().try_into().map_err(|_| Error::<T>::InvalidAccount)?;
+			Ok(Location::new(0, [AccountId32 { network: None, id: account_bytes }]))
+		}
```

Comment:
you can replace this fn with a bound on `AccountIdOf<T>: Into<Location>` as [convertors already exist](https://github.com/paritytech/polkadot-sdk/blob/0d644ca084f4f83271e30793d5618638016baf32/polkadot/xcm/src/v5/location.rs#L530C1-L541C2).

Reply:
Removed this, since is it no longer necessary in this pallet.